steel/docs/steel-history.md

```diff
@@ -0,0 +1,94 @@
+# Steel History
+
+## Why use Steel history?
+
+Steel commitments are necessary to ensure the integrity of view call data by confirming the block hash or the beacon block root, returned by the RPC provider, matches the on-chain available block information. Due to this necessity of checking against available on-chain block information, there is currently a age limit on view call data. When using the `blockhash` opcode, there is a context window of 256 blocks, meaning that the Steel proof is only valid for view call data within the last 256 blocks (approximately ~50 minutes with 12 second block time). When using L1 Ethereum, the [EIP-4788] beacon roots contract allows validation of the block root on the [beacon chain]. This technique extends the validation time to just over 24 hours. See the [Steel Commitments] page for further information.
+
+This age limit is a fundamental consequence of the way Ethereum based blockchains store relevant block data. For the developer, this means that **the view call data for a Steel execution proof can be a maximum of 24 hours old before the proof can not be verified on-chain**.  To use older view call data, the Steel library has a `history` feature. 
+
+## Overview
+
+The Steel history feature enables the developer to query view call state older than 24 hours while still using the same Steel commitment mechanism. This is done by separating the pinned block in Steel into two separate blocks: the execution block and the commitment block. 
+
+Without the Steel history feature enabled, the pinned block is *both* the block from which the view call state is retrieved and the block used for the Steel commitment:
+
+```rust
+// WITHOUT STEEL HISTORY
+// Create an EVM environment from that provider defaulting to the latest block.
+let mut env = EthEvmEnv::builder()
+    .provider(provider.clone())
+    .build()
+    .await?;
+
+// Preflight the call at the pinned block, default latest.
+let mut contract = Contract::preflight(args.token_contract, &mut env);
+```
+
+With the Steel history feature enabled, the developer must specify both the execution block *and* the commitment block:
+
+```rust
+/// WITH STEEL HISTORY
+let builder = EthEvmEnv::builder()
+    .provider(provider.clone())
+    .block_number_or_tag(args.execution_block);
+
+#[cfg(any(feature = "beacon", feature = "history"))]
```

Comment:
The history and the beacon feature were just added to the example so that the same example can be used to test all these different variants.
Do you think it is best to leave those cfg's in to match the example 100% or just remove the cfg lines to make the code snippet cleaner?

Reply:
I'd definitely suggest we remove the `cfgs` from the example. In a user application, we expect they will select an approach and chose one or these other, rather than having feature flags like this.

Reply:
I understand the reasoning for removing `cfgs`, and will keep that in mind for the future. Thanks both